src/components/SignUpForm/SignUpForm.tsx

```diff
@@ -41,66 +45,79 @@ export const SignUpForm = () => {
-      </Stack>
-
-      <fieldset className={styles.fieldset}>
-        <legend>{t('password')}</legend>
+    <>
+      <form className={styles.form} noValidate onSubmit={(event) => void handleSubmit(onSubmit)(event)}>
+        <Title className={styles.title} ta='center'>
+          {t('signUpTitle')}
+        </Title>
         <Stack gap={0}>
-          <PasswordInput
-            {...register('password')}
+          <TextInput
+            {...register('name')}
             className={styles.inputLabel}
-            label={t('enterPassword')}
+            label={t('name')}
             required
             radius='md'
-            id='password'
+            id='name'
           />
-          {passwordStrength && <PasswordStrengthMeter strength={passwordStrength} />}
-          <p className={styles.formErrors}>{errors.password?.message}</p>
+          <p className={styles.formErrors}>{errors.name?.message}</p>
         </Stack>
+
         <Stack gap={0}>
-          <PasswordInput
-            {...register('confirmPassword')}
+          <TextInput
+            {...register('email')}
             className={styles.inputLabel}
-            label={t('confirmPassword')}
+            label='Email'
             required
             radius='md'
-            id='password2'
+            id='email'
           />
-          <p className={styles.formErrors}>{errors.confirmPassword?.message}</p>
+          <p className={styles.formErrors}>{errors.email?.message}</p>
         </Stack>
-      </fieldset>
-      <Button className={styles.authBtn} disabled={!isDirty || !isValid}>
-        {t('signUpBtn')}
-      </Button>
-      <Text ta='center' mt='sm' className={styles.subtitle}>
-        {t('signInInquiry')}
-        <Link className={styles.link} href='/sign-in'>
-          {' '}
-          {t('signInLink')}
-        </Link>
-      </Text>
-    </form>
+
+        <fieldset className={styles.fieldset}>
+          <legend>{t('password')}</legend>
+          <Stack gap={0}>
+            <PasswordInput
+              {...register('password')}
+              className={styles.inputLabel}
+              label={t('enterPassword')}
+              required
+              radius='md'
+              id='password'
+            />
+            {passwordStrength && <PasswordStrengthMeter strength={passwordStrength} />}
+            <p className={styles.formErrors}>{errors.password?.message}</p>
+          </Stack>
+          <Stack gap={0}>
+            <PasswordInput
+              {...register('confirmPassword')}
+              className={styles.inputLabel}
+              label={t('confirmPassword')}
+              required
+              radius='md'
+              id='password2'
+            />
+            <p className={styles.formErrors}>{errors.confirmPassword?.message}</p>
+          </Stack>
+        </fieldset>
+        <Button className={styles.authBtn} disabled={!isDirty || !isValid}>
+          {t('signUpBtn')}
+        </Button>
```

Comment:
```suggestion
        <Button className={styles.authBtn} disabled={!isDirty || !isValid}>
          {t('signUpBtn')}
        </Button>
``` {loading ? (
        <div className={styles.spinnerWrapper}>
          <PawSpinner size={40} pawColor='var(--color-primary)' />
        </div>
      ) : (
        <Button className={styles.authBtn} disabled={!isDirty || !isValid}>
          {t('signUpBtn')}
        </Button>
      )}

      спинер как-то так можно добавить например

Reply:
Done